debian/rules

```diff
@@ -6,15 +6,16 @@
 # assume we only have one package
 PACKAGE_NAME:=`dh_listpackages`
 
-override_dh_systemd_enable:
-	dh_systemd_enable --name=matrix-synapse
-
-override_dh_installinit:
-	dh_installinit --name=matrix-synapse
+override_dh_installsystemd:
+	dh_installsystemd --name=matrix-synapse
```

Comment:
is it correct that `dh_installsystemd` will not install `matrix-synapse.default`? If so, we should probably remove it, and the reference from `matrix-synapse.service`.

Reply:
> people might be using it.

good point.

> This should probably go through some deprecation cycle.

Possibly, but if so, how do you suggest that we bring this to peoples' attention?

Maybe we should detect the presence of a (non-empty) `/etc/defaults/matrix-synapse` and emit some sort of warning in the postinst?

Reply:
> Possibly, but if so, how do you suggest that we bring this to peoples' attention?

In principle apt can show you the changelog entries on update (that's what it's therefore), but I think that's only on by default in testing and I vaguely remember hearing that this mechanism is broken at the moment under some circumstances anyway. This would also not reach people installing this via some automated systems, say config management, which will probably be the majority. 

I guess the safest would be to say this in the release announcement or upgrade guide as for other deprecations, too. I think it has a much greater chance of reaching users there than in the Debian changelog or via some warning by a postinst script. 

Also, it is not a strictly necessary to remove the `EnvironmentFile=` directive at all (even if it is somewhat ugly because it breaks up config some more and makes it impossible to glance the execution environment by looking at the unit file), since with the initial dash the service won't mind when it's not there. 

Generally I'd lean towards deprecating `/etc/default/matrix-synapse` in the upgrade guide for installations using the Debian packages and documenting that people who *want* to use this, will have to add a dropin config (like they already would need to get the unit dependencies right if they use postgres on the same machine or redis in case of worker setups), while maybe steering them towards using the config file instead.

Reply:
Yeah, that all sounds sensible.

maybe we just add a comment to the `.service` file saying `EnvironmentFile` is there for backwards compat is no longer recommended, and worry about properly deprecating it another time.

Reply:
Pushed a commit with a suggestion for this.